### PR TITLE
[Dispatch] Disable scatter fusion with producers

### PR DESCRIPTION
**Pull request by IanWood1**
Backends don't currently support scatter fusion and will silently compile incorrect code. This should be turned off in order to prevent backends from generating incorrect results. I don't think any users are running into this currently, but its best to keep it off for now. Similar to https://github.com/iree-org/iree/pull/19535 but for both `indices` and `updates`.

### CODE REVIEW

**Review by MaheshRavishankar — COMMENTED** [review]
Instead of this can we just disable this during dispatch region formation

**Comment by IanWood1** [comment]
I made the requested change. Additionally, instead of removing the tests, I modified them to check that no fusion occurs

**Review by MaheshRavishankar — APPROVED** [review]

File: compiler/src/iree/compiler/DispatchCreation/FormDispatchRegions.cpp [hunk 1]
```diff
@@ -651,7 +651,8 @@ isFusableWithProducer(OpOperand &operand,
   }
 
   // Don't fuse attention with it's producer
-  if (isa<IREE::LinalgExt::AttentionOp>(consumer)) {
+  // TODO: Enable scatter fusion when supported by backends.
```
Comment (hanhanW) [inline compiler/src/iree/compiler/DispatchCreation/FormDispatchRegions.cpp:654]: > Backends don't currently support scatter fusion and will silently compile incorrect code. 

Is there an issue for the TODO or your comment in the PR description?

Comment (hanhanW) [reply compiler/src/iree/compiler/DispatchCreation/FormDispatchRegions.cpp:654]: I found an issue about fusion (https://github.com/iree-org/iree/issues/19091), but not an issue for correctness. I mainly wonder if we should allocate someone from codegen side to fix the issue.

Comment (IanWood1) [reply compiler/src/iree/compiler/DispatchCreation/FormDispatchRegions.cpp:654]: I think either @qedawkins or @MaheshRavishankar were going to look at this at some point. But I don't think there is a dedicated github issue for the correctness problem.

**Pull request merged**